hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java

```diff
@@ -37,13 +49,34 @@
     private static final long ALERT_REPEAT_MILLIS = Parameter.longValue("spawn.job.alert.repeat", 5 * 60 * 1000);
     private static final long ALERT_DELAY_MILLIS = Parameter.longValue("spawn.job.alert.delay", 60 * 1000);
 
+    private enum AlertAction {CREATE_OR_UPDATE, DELETE, NO_OP}
+
+    private static final long MAX_RUNTIME = Parameter.longValue("alert.auto.runtime.max", 480);
+    private static final long RUNTIME_PADDING = Parameter.longValue("alert.auto.runtime.padding", 30);
+    private static final long REKICK_PADDING = Parameter.longValue("alert.auto.rekick.padding", 30);
+    private static final String EXTRA_DESCRIPTION =
+            "\nThis alert was automatically generated by Hydra. Use job settings (Basic Hydra Alerts/Pagerduty) to " +
+            "manage. Do not modify directly.";
+
     private final JobAlertRunner jobAlertRunner;
     private final SpawnDataStore spawnDataStore;
+    private final GroupManager groupManager;
+
+    @VisibleForTesting public JobAlertManagerImpl(
+            Spawn spawn,
+            JobAlertRunner jobAlertRunner,
+            ScheduledExecutorService scheduledExecutorService) {
+        this.spawnDataStore = spawn.getSpawnDataStore();
+        this.jobAlertRunner = jobAlertRunner;
+        this.groupManager = spawn.getGroupManager();
+        this.scheduleAlertScan(scheduledExecutorService);
+    }
 
     public JobAlertManagerImpl(Spawn spawn, ScheduledExecutorService scheduledExecutorService) {
         this.spawnDataStore = spawn.getSpawnDataStore();
-        this.jobAlertRunner = new JobAlertRunner(spawn, areAlertsEnabled());
-        scheduleAlertScan(scheduledExecutorService);
+        this.jobAlertRunner = new JobAlertRunner(spawn, this.areAlertsEnabled());
+        this.groupManager = spawn.getGroupManager();
+        this.scheduleAlertScan(scheduledExecutorService);
```

Comment:
I meant this constructor can call the other overloaded constructor - initialization code is the same with the exception of jobAlertRunner, which the other constructor takes as a parameter